### ISSUE TITLE
Extended Ban Problem

### ISSUE THREAD

**Issue by ComputerTech312** [ComputerTech312/Mute-Script#1]
Some networks dont support Extended bans 
e.g +b ~q:$nick2mute!*@* 

so in this case you could either do

putserv "mode $chan +b :$nick2mute!*@* "

or

putserv "mode $chan quiet:$nick2mute!*@* "

and do for the opposite for the undoing of it

mode $chan -b :$nick2unmute!*@* 

mode $chan unquiet:$nick2unmute!*@*

**Comment by KoraggKnightWolf** [comment]
It is not only that some do not support extbans but they support other syntaxes. The fact that nick!*@* bans are trivially bypassed via /NICK is another issue. Some of the most/more common ones:
InspIRCd: +b m:nickname!*@*
UnrealIRCd: +b ~q:nickname!*@*
Charybdis/ircd-seven: +q nick!*@*

Perhaps having an extban type configuration would be of use e.g. extban-type insp|unreal|chary 
(Side note regarding mutes do not stop nick changes: Chary/seven also stop those with +q but as most networks use Insp or Unreal muting by nick is suboptimal).
Perhaps a mute on *!*user@host is an ideal middleground (* before user to catch ~ or no ~)

**Comment by ComputerTech312** [comment]
Yep, good point Koragg this was the first script i published on github, i am updating it with Two things:
Devoice the user first
Set ban on *!user@host (as you said above)
as not all unreal networks have the module installed that takes care of banning voiced users, so i guess devoicing them first does the trick heh :)
